src/components/shared/Rating/index.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+
+import styled from 'styled-components/native';
+
+const COLOR: {
+  [key: string]: string;
+} = {
+  WHITE: '#ffffff',
+  BACKGROUND: '#EFF0F1',
+  ACTIVE: '#FFB402',
+  INACTIVE: '#BDBDBD',
+};
+interface Props {
+  total: number;
+  value: number;
+  onChange: Function;
+  readonly: boolean;
+  disabled: boolean;
+}
+interface ChildProps {
+  total: number;
+  on: boolean;
+  focus: boolean;
+}
+const ContainerWrapper = styled.View`
+  width: ${(props: Props): number => props.total * 30};
+  height: 30px;
+  background-color: ${COLOR.BACKGROUND};
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+  opacity: ${(props: Props): number => (props.disabled ? 0.3 : 1)};
+`;
+const StarComponent = styled.TouchableOpacity`
+  width: ${(props: ChildProps): string =>
+    props.focus ? `${(100 / props.total) * 1.02}%` : `${100 / props.total}%`};
+  height: ${(props: ChildProps): string => (props.focus ? '102%' : '100%')};
+  background-color: ${(props: ChildProps): string =>
+    props.on ? COLOR.ACTIVE : COLOR.INACTIVE};
+  border: 1px solid white;
+`;
+
+function Rating(props: Props): React.ReactElement {
+  const [value, setValue] = useState(0);
+  const [stars, setStars] = useState(new Array(props.total).fill(false));
+  const [focus, setfocus] = useState(false);
+  const _onChangeValue = (): void => {
+    const result = stars.fill(false);
+    let i = 0;
+    while (i < value) {
+      result[i] = true;
+      i++;
+    }
+    setStars(result);
+    if (props.onChange) props.onChange({ stars: value });
```

Comment:
It can be `props.onChange && props.onChange({ stars: value });`. But current code is also Good.